File: src/data/courses/dsa/stacks/stack-core-operations/implement-stack-operations/code/python/implement_stack.py
class Stack:
    def __init__(self):
        self.arr = []
    
    def push(self, val):
        self.arr.append(val)
    
    def pop(self):
        return self.arr.pop()
    
    def peek(self):
        return self.arr[-1]
    
    def is_empty(self):
        return len(self.arr) == 0
    
    def size(self):
        return len(self.arr)

File: src/data/courses/dsa/stacks/stack-core-operations/implement-stack-operations/code/cpp/ImplementStack.cpp
class Stack {
    int* arr;
    int top;
    int capacity;
public:
    Stack(int size) {
        arr = new int[size];
        capacity = size;
        top = -1;
    }
    
    void push(int val) {
        arr[++top] = val;
    }
    
    int pop() {
        return arr[top--];
    }
    
    int peek() {
        return arr[top];
    }
    
    bool isEmpty() {
        return top == -1;
    }
    
    int size() {
        return top + 1;
    }
};
